### DISCUSSION TITLE
Application command integration

### DISCUSSION THREAD

**Discussion by bsian03** [abalabahaha/eris#1266] [Ideas]
With the introduction of the message content priveleged intent coming next year, Discord is asking developers to use application commands, unless they have a use case where they will need the intent for their bot to function.

We already have a PR which allows users to create slash commands, #1210

We also have our own in-house command client using message content which doesn't interact with Discord. 
The question is whether we should do a similar thing for application commands, and if we do, whether the internals would pretty much be the same, how we handle Discord limits, whether we should rework the CommandClient as a whole, how we handle Discord restrictions and the slightly more complex setup, as well as possible duplicate connections(?)

Feel free to throw your opinions/ideas on how we should move forward with this.

**Comment by coolcalcacol** [comment]
I guess we should try and figure out the main usecase for the command client, and whether or not it's expected to be used in servers that will actually have these restrictions applying to them, and take it from there?

**Reply by JustCat80** [reply]
we could also consider doing both, and just having it as an option to toggle them on and off?
